.github/workflows/mac-compatibility.yml

```diff
@@ -0,0 +1,38 @@
+# GitHub Actions for genCMPClient
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Siemens AG, 2021-2024
+
+name: macOS
+
+on: push
+
+jobs:
+  cmake:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: make
+        run: |
+          make -f Makefile_v1
+          ./cmpClient -help
+          make -f Makefile_v1 clean_all
+      - name: cmake
+        run: |
+          cmake .
+          make
+          ./cmpClient -help
+
```

Comment:
I think we can leave this variation out since it is already checked by the normal build CI
and there is no special OpenSSL dependency here.